draft-ietf-tls-esni.md

```diff
@@ -1025,9 +1012,9 @@ The backend server MUST NOT perform this operation if it negotiated TLS 1.2 or
 below. Note that doing so would overwrite the downgrade signal for TLS 1.3 (see
 {{RFC8446}}, Section 4.1.3).
 
-The "ech_is_inner" is expected to have an empty payload. If the payload is
-non-empty (i.e., the length of the "extension_data" field is non-zero) then the
-backend server MUST abort the handshake with an "illegal_parameter" alert.
+The payload of "encrypted_client_hello" is expected to be a `ClientECH` with
+`ClientECH.type` is `inner`. If this is not the case, the backend server MUST
+abort the handshake with an "illegal_parameter" alert.
```

Comment:
I think we can delete this paragraph:

The old text wasn't talking about inner vs outer but about the payload of the extension. But that's part of the syntax of the extension so it'd be decode_error and also already captured in the extension's syntax.

The new text seems to be talking about inner vs outer, but we've already captured that in the precondition for this section. You only go into the backend server section if it's already an inner variant.

Thus, I think we can trim this. :-)

Reply:
Good call :)